llvm/lib/Target/RISCV/RISCVLoadStoreOptimizer.cpp

```diff
@@ -0,0 +1,371 @@
+//===----- RISCVLoadStoreOptimizer.cpp ------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Bundle loads and stores that operate on consecutive memory locations to take
```

Comment:
What is the reason to have a pass instead of doing this in the scheduler?

Reply:
Isn't it creating a new paired instruction by combining two instructions? How can you do that in the scheduler?

Reply:
Sorry, I thought this was doing clustering, not combining. My bad.